### ISSUE TITLE
Electricity Pads

### ISSUE THREAD

**Issue by ncrecc** [BEEmod/BEE2-items#2179]
The electricity pads from Portal: Still Alive.
![](https://image.prntscr.com/image/PoPjlerDSaSF2Kg8YCpz3w.png)
These would periodically turn on and off. While they're on, they kill players, and only players.
For an example, play Level 5 of [the PC port](http://www.moddb.com/mods/portal-still-alive-pc) or Chamber 07 of Portal: Prelude. Prelude, however, has them textured with a yellow flash, and Still Alive has them textured as blue lightning bolts.

**Comment by TeamSpen210** [comment]
Indeed, I've wanted to add those. Perhaps you place them on one surface, and they clone themselves to the other wall. You'd link them like cutout tiles or triggers. They would have a timer option, to change how long it stays in each mode.

**Comment by TeamSpen210** [comment]
I'll also want to do some VScripts, spawn in some env_beams which randomly fire throughout the field region while it's on.

**Comment by Luke18033** [comment]
I think it already does that? Also, Portal: Still Alive uses a (custom?) texture which slowly fades in to indicate the field is turning on, which we'd want to include.

**Comment by TeamSpen210** [comment]
It does have beams, but I'd want to use VScript, so they connect to the two sides. The texture is custom, but we can replace it with the Light Bridge edge texture. Pretty much the same.

**Comment by Byzarru** [comment]
This should also be the item that electrifies water in the BP pack.

**Comment by Endermage77** [comment]
I'd personally like to be able to manually toggle this, and control it with a separate timer item thing.

**Comment by TeamSpen210** [comment]
Perhaps in ∞ mode it'll just turn on and off directly.

**Comment by Endermage77** [comment]
Why not add a timer that can automatically toggle itself like this?

**Comment by TeamSpen210** [comment]
I'd want to do that too as well.

**Comment by Endermage77** [comment]
Oh, okay.

**Comment by Nikita74Russian** [comment]
If you have timer set to any time, you can press the input to shutdown it permanently. Or if pedestal button used, you can shutdown it for a timelimit

**Comment by SomePeople27** [comment]
You could set it to Start Reversed to allow it to dissolve things.

**Comment by SomePeople27** [comment]
it wont dissolve player ofc .-.

**Comment by SomePeople27** [comment]
just dissolve cubes and stuff.

**Comment by TeamSpen210** [comment]
No, how do you tell the difference between a cube version and a player version?

**Comment by Nick2007Portal** [comment]
I don't understand what are you trying to say

**Comment by King0191** [comment]
How do you let the player know the difference between a Object Fizzling Pad. or a Kill Trigger Pad?
Signage? Yeah, sure but not everyone LOOKS at Signage.
Different colors? Thats Trial and Error.
You need an easy and quick way to tell them apart.

**Comment by Nick2007Portal** [comment]
It will do both - kill players and dissolve objects. Electric bolts will be red.

**Comment by King0191** [comment]
Still isn't going to communicate it well enough. This object doesn't need to fizzle objects. If you want to prevent bringing items through it. Place Fizzlers on each end,

**Comment by Nick2007Portal** [comment]
Won't do. Fizzlers don't fit that thin.

**Comment by Nick2007Portal** [comment]
If not, then what other you offer me? Different sounds, colour? 
CHECK AND MATE!

**Comment by Endermage77** [comment]
It's quite simple:
If TeamSpen doesn't feel that it can be distinguished easily from a regular one, then it shall not pass any further than a Balrog past Gandalf.

**Comment by King0191** [comment]
Sounds, Color, all of that is irrelevant. Electricity Pads were never made to fizzle objects. Why should we change that? We have so many other objects that can fizzle them. That at this point items that can fizzle this is just an "Oh... Another object that does this?"

**Comment by ZipZoptheCat** [comment]
There's alsways the standing fizzler. You could put it outside the end of the electricity pad.

**Comment by TeamSpen210** [comment]
Another component is what makes logical sense. Electrical shocks will obviously kill people. For an inanimate block, it's not going to do anything. So those properties are identifiable by players well before they actually try them out. Good test elements show the player what they will do. Fizzlers glow and speed up their effect when you bring cubes near them, to ahow the impending doom. It makes sense that a solid Light Bridge would support gel on it, and that it would fall off when disabled.

**Comment by G-Guy** [comment]
I know I am responding in an 3 year old thread but would we possibly see those in 4.38 build or anytime in future?

**Comment by FenceAKAGlasnost** [comment]
They also exist in Portal Prelude, I think

**Comment by Luke18033** [comment]
If this ever gets added it should probably also have a "forgiving" mode like laserfields, where it damages and shoves you out if you're standing near the edge and only instakills you if you try to force through or stand in the middle

**Comment by LincoCompany00** [comment]
It should also receive inputs to disable it as well.

**Comment by LincoCompany00** [comment]
I made up an old aperture version. It would be very similar to the Old Aperture fizzler emmitters (this is the best picture I've got, sorry)
![Image](https://github.com/user-attachments/assets/3dbdff2a-05d0-45fd-87f0-c15baff0e482)
Arranged in a 4×4 grid, kind of like this. For each voxel it takes up, the metal bar frame wouldn't extend for the entire length of the field, but would repeat, like of like tall fizzlers do.  
![Image](https://github.com/user-attachments/assets/37796dba-7afd-420a-b99e-7ec380d824f3)
However, so it still resembles the Clean virant, the emmitter things would be below the bars, like this:
![Image](https://github.com/user-attachments/assets/87515500-8e01-4fa7-8dfe-eefa144263a2)
It's effect would be very similar to the Old Aperture laserfield, with arcs of electricity between the emmitter things, but it would be purple or yellow.
I'm trying to explain it the best I can... do you understand? I don't have modeling software, or I'd make it myself.

**Comment by TeamSpen210** [comment]
That makes sense yes, though I'm not sure how good perf wise it'd be to have so many. Probably fine nowadays.

**Comment by vrad-exe** [comment]
Eh I don't know how I feel about that design. I feel like it would make more sense to make the design look like exposed wiring or something. Maybe the emitters could just be at the corners or edges.

**Comment by Nuclearxpotato** [comment]
Just to add onto the topic: these are one of my favourite "introduced" elements (which technically includes the Hardlight Bridge / Blue Wall) because it perfectly matches the oddly murderous personality of the testing elements in Portal 1 like the HEP... it's as iconic of a P1 element to me. I primarily build in the P1 style so I'm a little biased.

**Comment by LincoCompany00** [comment]
The Blue Plasma Field, or what you call the  "Hard Light Bridge," is the same as the Force Deflection Field. Turrets can fire though it, so it's not a light bridge anyway. It's been suggested that the Hard Light Bridge texture be changed to this is in P1 style, but unfortunately it's hard-coded into the game. So this won't be implemented, unfortunately.
 The Laserfield is in there, though!

**Comment by LincoCompany00** [comment]
Personally I like the Portal: Still Alive blue  lighting bolt effect and don't really like the Portal: prelude effect (or what I assume is the Portal: Prelude effect), as Still Alive's seems better, while Prelude's single bolt zapping effect just seems it seems more like a cheap effect and doesn't make a ton of sense.

Still Alive's seems more Portal 1-y than Prelude's, so P1 style could be the Still Alive blue lightning bolt effect, Clean style could use the yellow Portal: Prelude style one (or a yellow version of Still Alive's effect). 
Old Aperture would use the Portal: Prelude's yellow flash effect, but with multiple of the lightning bolts throughout the field— as Portal Prelude's style makes more sense for Old Aperture's exposed wiring.

**Comment by LincoCompany00** [comment]
Here's a quick drawing I made to further "demonstrate:"
![Image](https://github.com/user-attachments/assets/5463f05a-7f24-4fe9-b772-69e34b8eeda4)

**Comment by LincoCompany00** [comment]
I just had another thought on how the item could work— It's a fizzler-class item, which extends to the opposite wall, but it has no rotation handle and it's editor model is a full square.
You place one at each corner and connect them with antlines, and it will generate the electricity pads!

**Comment by TeamSpen210** [comment]
That's an interesting method, though it might be confusing - the field visual is meaningless, and you should be able to expand it to a rectangle or other shapes. We'd still need a rotation handle even if 4-way, since otherwise you'd be locked to one direction of expansion.
